### DISCUSSION TITLE
EventBus module

### DISCUSSION THREAD

**Discussion by olivermrbl** [medusajs/medusa#2838] [Feature Requests]
### What
Introduce an EventBus abstraction allowing businesses to use any PubSub provider to power the event's infrastructure in Medusa

### Motivation
With our current architecture, you are forced to have Redis as part of your stack if you want to leverage the events system in Medusa. Ideally, the events system should not be tied to a specific provider, such that if your existing infrastructure contains a PubSub provider different from Redis (e.g. RabbitMQ), you can use it to power Medusa's events system.

**Comment by kawpii** [comment]
@olivermrbl This is great!, thanks for doing this!
